rust/src/rfb/rfb.rs

```diff
@@ -0,0 +1,837 @@
+                                    current_transaction.tc_failure_reason = Some(request);
+                                }
+                                _ => {
+                                    return false;
+                                }
+                            }
+
+                            return false;
+                        }
+                        Err(nom::Err::Incomplete(_)) => {
+                            self.request_buffer.extend_from_slice(current);
+                            return true;
+                        }
+                        Err(_) => {
+                            return false;
+                        }
+                    }
+                }
+                parser::RFBGlobalState::TCServerInit => {
+                    match parser::parse_server_init(current) {
+                        Ok((rem, request)) => {
+                            current = rem;
+                            self.state = parser::RFBGlobalState::Message;
+
+                            match self.get_current_tx() {
+                                Some(current_transaction) => {
+                                    current_transaction.tc_server_init = Some(request);
+                                    // connection initialization is complete and parsed
+                                    current_transaction.complete = true;
+                                }
+                                _ => {
+                                    return false
+                                }
+                            }
+                        }
+                        Err(nom::Err::Incomplete(_)) => {
+                            self.request_buffer.extend_from_slice(current);
+                            return true;
+                        }
+                        Err(_) => {
+                            return false;
+                        }
+                    }
+                }
+                parser::RFBGlobalState::Message => {
+                    //todo implement RFB messages, for now we stop here
+                    return false;
+                }
+                _ => {
+                    SCLogDebug!("Invalid state for response");
+                    return false;
+                }
+            }
+        }
+    }
+
+    fn tx_iterator(
+        &mut self,
+        min_tx_id: u64,
+        state: &mut u64,
+    ) -> Option<(&RFBTransaction, u64, bool)> {
+        let mut index = *state as usize;
+        let len = self.transactions.len();
+
+        while index < len {
+            let tx = &self.transactions[index];
+            if tx.tx_id < min_tx_id + 1 {
+                index += 1;
+                continue;
+            }
+            *state = index as u64;
+            return Some((tx, tx.tx_id - 1, (len - index) > 1));
+        }
+
+        return None;
+    }
+}
+
+/// Probe to see if this input looks like a request or response.
+fn probe(input: &[u8]) -> bool {
```

Comment:
Should not this be made as a fixed pattern instead of a probing function ? ie calling `AppLayerProtoDetectPMRegisterPatternCI`

Reply:
Yes! Just changed that and will push it in the next PR. Thanks.

Reply:
Thanks